Promote `AdminKubeconfigRequest` featuregate to Beta (#5)
shafeeqes/gardener: docs/proposals/16-adminkubeconfig-subresource.md

        client-key-data: LS0tLS1CRUd...
```

> The `AdminKubeconfigRequest` feature gate (enabled by default starting from `v1.39`) would enable this API in the `gardener-apiserver`. The old `{shoot-name}.kubeconfig` would be kept, but deprecated and removed in the future.

Reviewer: ```suggestion
The `AdminKubeconfigRequest` feature gate (enabled by default starting from `v1.39`) enables the above mentioned API in the `gardener-apiserver`. The old `{shoot-name}.kubeconfig` is kept, but deprecated and will be removed in the future.
```

Reviewer: In the main PR we can discuss whether we want to remove the 'old' {shoot-name}.kubeconfig` in this PR. I will need to have a deeper look in the code though before that.
